pkg/config/mesh/networks_watcher.go

```diff
@@ -32,18 +33,30 @@ type NetworksHolder interface {
 	PrevNetworks() *meshconfig.MeshNetworks
 }
 
+// WatcherHandlerRegistration will be returned to caller to remove the handler later.
+type WatcherHandlerRegistration interface{}
+
+type watchHandler struct {
+	WatcherHandlerRegistration
```

Comment:
I cannot get what is the interface for

Reply:
This design is inspired by `AddEventHandler(handler ResourceEventHandler) (ResourceEventHandlerRegistration, error)` and `RemoveEventHandler(handle ResourceEventHandlerRegistration) error` of SharedInformer. 

I know there are improved spaces. Need more suggestions.